azavea/echo-locator
# Only calculate active route

## Overview

Improve performance on navigation back to list from details by only calculating the active route to draw.

All route paths were being calculated before, but only the active one was styled as visible. As the styling for the active route only changes when all routes are recalculated, and the non-active routes are never displayed, it is not necessary to compute them all at once.


### Demo

0.8 ms was spent in `getStopPositions` here, down from 138 ms in #166 performance log.
![image](https://user-images.githubusercontent.com/960264/56831863-50871c80-6838-11e9-96b3-f0418aa518a0.png)


## Testing Instructions

 * Navigation back from details to list should have no visible map redraw delay
 * Active route display on sidebar card hover or detail select should still function as expected


Closes #166.
